src/test/ui/or-patterns/basic-switch.rs

```diff
@@ -0,0 +1,33 @@
+// Test basic or-patterns when the target pattern type will be lowered to a
+// `SwitchInt` (an `enum`).
```

Comment:
I find it weird (as in, I bet something's misnamed) that `basic-switch.rs` has a comment saying that it tests the `SwitchInt` case, while `basic-switchint.rs`  has a comment saying that it tests the `Switch` case...